### PR TITLE
convert to the base minus 2 of a number

### LINKED ISSUE #9588: Convert to Base -2
### Feature description

Found a solution to this problem.

I would like to solve this issue.

### PR DESCRIPTION
**Pull request by haxkd**
### Describe your change:

Fixes: #9588

Added maths/base_neg2_conversion.py.

* [x] Add an algorithm?
* [] Fix a bug or typo in an existing algorithm?
* [] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #9588".

### CODE REVIEW

**Comment by cclauss** [comment]
Please change the title of this pull request so we know what it does.

File: maths/base_neg2_conversion.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        100
+        >>> decimal_to_negative_base_2(7)
+        11011
+    """
+    if num == 0:
+        return 0
+    ans = ""
+    while num != 0:
+        rem = num % -2
+        num = num // -2
```
Comment (cclauss) [inline maths/base_neg2_conversion.py:27]: Let's use https://docs.python.org/3/library/functions.html#divmod instead

File: maths/base_neg2_conversion.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+        ans = str(rem) + ans
+
+    return int(ans)
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
```
Comment (cclauss) [inline maths/base_neg2_conversion.py:40]: Please revert the file path change below.

File: maths/base_neg2_conversion.py (hunk 1 above)
Comment (haxkd) [reply maths/base_neg2_conversion.py]: done

**Review by cclauss — APPROVED** [review]

**Pull request merged**